src/machines/lesson-player-machine.ts

```diff
@@ -27,12 +28,13 @@ export type PlayerStateEvent =
   | {type: 'PLAY'}
   | {type: 'PAUSE'}
   | {type: 'LOADED'}
-  | {type: 'LOADED'; lesson: any; viewer: any}
+  | {type: 'LOADED'; lesson: any; viewer: any; isIncomingAnonViewer?: boolean}
```

Comment:
You added `isIncomingAnonViewer` to the type, but it isn't set or used within the machine. It looks like the lesson page gets by fine with it as component local state, so I think you can ditch the type change here.